src/foam/core/XMLSupport.java

```diff
@@ -115,10 +115,13 @@ public static void copyFromXML(X x, FObject obj, XMLStreamReader reader) throws
         eventType = reader.next();
         switch ( eventType ) {
           case XMLStreamConstants.START_ELEMENT:
-            var prop = (PropertyInfo) cInfo.getAxiomByName(reader.getLocalName());
-            if ( prop != null ) {
-              prop.copyFromXML(x, obj, reader);
-              prop = null;
+            List<PropertyInfo> properties  = cInfo.getAxiomsByClass(PropertyInfo.class);
```

Comment:
This will be very slow because it has to loop over every PropertyInfo for every property parsed. Making it an O(n^2) instead of O(n). I'll add an   "Object      getAxiomByNameOrShortName(String name)" method to ClassInfo to make this easier and faster to do.